Basic/APIDiorama/Assets/UseCaseSamples/ProximityChecks/Scripts/ColorManager.cs

```diff
@@ -0,0 +1,70 @@
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Manages the color of a Networked object
+    /// </summary>
+    public class ColorManager : NetworkBehaviour
+    {
+
+        NetworkVariable<Color32> m_NetworkedColor = new NetworkVariable<Color32>();
+        Material m_Material;
+        ProximityChecker m_ProximityChecker;
+
+        void Awake()
+        {
+            m_Material = GetComponent<Renderer>().material;
+            m_ProximityChecker = GetComponent<ProximityChecker>();
+        }
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            if (IsClient)
+            {
+                OnClientColorChanged(m_Material.color, m_NetworkedColor.Value);
+                m_NetworkedColor.OnValueChanged += OnClientColorChanged;
+            }
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            base.OnNetworkDespawn();
+            if (IsClient)
+            {
+                m_NetworkedColor.OnValueChanged -= OnClientColorChanged;
+            }
+        }
+
+        void Update()
+        {
+            if (!IsClient
+            || !m_ProximityChecker.LocalPlayerIsClose)
+            {
+                return;
+            }
+
+            if (Input.GetKeyUp(KeyCode.E))
+            {
+                OnClientRequestColorChange();
+            }
+        }
+
+        void OnClientRequestColorChange()
+        {
+            OnServerChangeColorServerRpc();
+        }
+
+        [ServerRpc(RequireOwnership = false)]
```

Comment:
there seems to be a lesson here about ownership and I didn't see IET text about this. It'd be great to have code comments about the pros but also the dangers of setting this attribute

Reply:
> there seems to be a lesson here about ownership and I didn't see IET text about this

We can't copy-paste the whole documentation here, it's a matter of balance between the length of tutorial and the core concepts (and there are [official guidelines](https://docs.google.com/document/d/1WjeVchvdUf2HbnUPpi_iu9XYhirAqi_S-foxipIe0Ek/edit) about these). In the NetworkVariable tutorial we explain the main concepts, and then we redirect users to the documentation to "learn more".

> It'd be great to have code comments about the pros but also the dangers of setting this attribute

The comments about pros and cons are in the documentation, I can add a small code comment to redirect there